Messages Pallet Documentation Improvement (#1981)
frequency-chain/frequency: pallets/messages/README.md

# Messages Pallet

Stores block-indexed data for a Schema using `OnChain` or `IPFS` payload location.

## Summary

Messages stores metadata and message payload data for a set Schema.

> Use of the Message pallet payloads is designed for time-centric data.

Author: Merging for now, but let me know if you have thoughts/better ideas and we can tweak it in one of the other docs PRs.

Discovery is via the `MessagesInBlock` on new blocks or requesting messages from a block.
Retrieval is via RPC.

### Metadata vs Payload

Messages have both metadata and payloads.

> The payload should always match the data structure or the messages is considered invalid.

Reviewer: Should read "or the message is considered invalid" or "or the messages are considered invalid"

Author: ```suggestion
The payload should always match the data structure or the message is considered invalid.
```

> The metadata is the Block Number, Schema Id, and other data to help discover and organize the payload information.

Reviewer: Would suggest "and other data useful for discovering and organizing the payload information".

Author: ```suggestion
The metadata is the Block Number, Schema Id, and other data useful for discovering and organizing the payload information.
```

### Payload Options

> - `IPFS`: Storage of the CID and length of the file on IFPS

Reviewer 2: ```suggestion
- `IPFS`: Storage of the CID and length of the file on IPFS
```

> - `IPFS`: Storage of the CID and length of the file on IPFS

Reviewer 3: period

Author: Removing periods elsewhere. I've been going without most places, and the one below slipped through with the original statement having two parts.

> - `OnChain`: Storage of the entire payload data. Usually for tiny payloads only.

Reviewer 3: not a big deal but what does tiny mean?

Author: Updated to:
`- `OnChain`: Storage of the entire payload data, usually for sub-256 byte payloads`

Author: Merging for now, but let me know if you have thoughts/better ideas and we can tweak it in one of the other docs PRs.

> ### Message Ordered
> 
> Messages are ordered by block number, but the have a set order based on transaction order in the block.

Reviewer 2: ```suggestion
### Message Ordering

Messages are ordered by block number, and within each block have a set order based on transaction order in the block.
```

> Messages are ordered by block number, and within each block have a set order based on transaction order in the block.

Reviewer 3: Maybe:
Messages are ordered by block number, and within each block, they follow a specific order based on their transaction sequence within that block.

This order is immutable.

### Actions

The Messages pallet provides for:

> - Adding messages for a given schema

Reviewer 3: minor: but sometimes there is a period at the end of these and sometimes there is not.
